app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectViewModel.kt

```diff
@@ -276,28 +293,38 @@ class WalletConnectViewModel(application: Application) : AndroidViewModel(applic
     }
 
     fun prettyPrintJson() {
+
         binder?.getSessionRequestParams()?.let { params ->
             val strategy: ExclusionStrategy = object : ExclusionStrategy {
                 override fun shouldSkipField(f: FieldAttributes): Boolean {
                     return f.name == "payload" || f.name == "schema"
                 }
+
                 override fun shouldSkipClass(clazz: Class<*>?): Boolean {
                     return false
                 }
             }
-            val gson = GsonBuilder().setPrettyPrinting().addSerializationExclusionStrategy(strategy).create()
             params.payloadObj = params.parsePayload()
             params.payload = ""
             if (params.payloadObj != null && params.payloadObj?.message != null && params.schema != null) {
                 CoroutineScope(Dispatchers.IO).launch {
-                    val jsonMessage = App.appCore.cryptoLibrary.parameterToJson(ParameterToJsonInput(params.payloadObj!!.message, params.payloadObj!!.receiveName, params.schema!!, null))
+                    val jsonMessage = App.appCore.cryptoLibrary.parameterToJson(
+                        ParameterToJsonInput(
+                            params.payloadObj!!.message,
+                            params.payloadObj!!.receiveName,
+                            params.schema!!,
+                            null
+                        )
+                    )
                     if (jsonMessage != null) {
-                        params.message = jsonMessage.replace("\"", "")
+                        params.message = jsonMessage.replace("\\/", "")
+                        jsonPretty.postValue(params.message!!.prettyPrint())
+                    } else {
+                        jsonPretty.postValue("")
                     }
-                    jsonPretty.postValue(gson.toJson(params).replace("payloadObj", "payload"))
                 }
             } else {
-                jsonPretty.postValue(gson.toJson(params).replace("payloadObj", "payload"))
+                jsonPretty.postValue("")
```

Comment:
Why is this like this? Doesn't this else branch indicate an error?

Reply:
Yes, on error an empty value is returned and the preview is removed from the UI